sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Distinct.java

```diff
@@ -78,26 +87,46 @@
    */
   public static <T, IdT> WithRepresentativeValues<T, IdT> withRepresentativeValueFn(
       SerializableFunction<T, IdT> fn) {
-    return new WithRepresentativeValues<T, IdT>(fn, null);
+    return new WithRepresentativeValues<>(fn, null);
+  }
+
+  private static <T> void validateWindowFn(PCollection<T> in) {
+    WindowingStrategy strategy = in.getWindowingStrategy();
+    checkArgument(strategy.getWindowFn().isNonMerging()
+      || (strategy.getTrigger().getClass().equals(DefaultTrigger.class)
+        && strategy.getAllowedLateness().isEqual(Duration.ZERO)),
+        "Dinstinct does not support non-merging windowing strategies, except when"
```

Comment:
On the error message, `String.format("%s does not support ...", Distinct.class.getSimpleName())`. IDE-friendly and immune to typos like the one here :-)